src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerFhirDataStore.cs

```diff
@@ -336,22 +336,55 @@ public async Task BulkUpdateSearchParameterIndicesAsync(IReadOnlyCollection<Reso
                     sqlCommandWrapper,
                     _bulkReindexResourcesTvpGeneratorVLatest.Generate(resources.ToList()));
 
-                try
+                if (_schemaInformation.Current >= SchemaVersionConstants.BulkReindexReturnsFailuresVersion)
                 {
-                    await sqlCommandWrapper.ExecuteNonQueryAsync(cancellationToken);
+                    // We will reindex the rest of the batch if one resource has a versioning conflict
+                    using (SqlDataReader sqlDataReader = await sqlCommandWrapper.ExecuteReaderAsync(CommandBehavior.SequentialAccess, cancellationToken))
+                    {
+                        if (!sqlDataReader.Read())
+                        {
+                            return;
+                        }
+
+                        int numberOfVersionConflicts = sqlDataReader.GetInt32(0);
+
+                        // TODO: Return total number of successfully reindexed resources.
+                        int numberOfDeletedResources = sqlDataReader.GetInt32(1);
```

Comment:
TBD in [AB#83909](https://microsofthealth.visualstudio.com/Health/_workitems/edit/83909).